doc/man3/X509_NAME_get_index_by_NID.pod

```diff
@@ -79,7 +79,8 @@ return the index of the next matching entry or -1 if not found.
 X509_NAME_get_index_by_NID() can also return -2 if the supplied
 NID is invalid.
 
-X509_NAME_entry_count() returns the total number of entries.
+X509_NAME_entry_count() returns the total number of entries, and 0
+and -1 for failure.
```

Comment:
It can only ever actually return 0 on failure I think. The one case where the underlying `sk_X509_NAME_ENTRY_num` can return -1 can never be hit because `X509_NAME_entry_count()` already returned 0 before calling  `sk_X509_NAME_ENTRY_num`. This is a bit brittle and the behaviour could change if `sk_X509_NAME_ENTRY_num()` were ever to return -1 in different circumstances. It might be a good idea to update `X509_NAME_entry_count()` to enforce "only return 0 on failure" a bit more strongly.